hybrid/compartment_examples/inter_comp_call_secure/shared.S

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2021 The CapableVMs "CHERI Examples" Contributors.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#include "main.h"
+
+.global comp_f_fn
+.global comp_g_fn
+
+.text
+.balign 4
+
+/* The compartment switch function. Expects compartment information to be
+ * stored in memory (defined by the capability stored in register `c19`).
+ * Performs a compartment switch based on the id saved in `CID_EL0` (currently
+ * just an integer).
+ */
+.global switch_compartment
+.type switch_compartment, "function"
+switch_compartment:
+
+    // Store entering compartment's DDC, and move to memory containing
+    // compartment info
+    mrs       c2, DDC
+    // Expect switcher DDC in c19
+    msr       DDC, c19
+
+    // Get compartment to switch to data
+    mrs       c10, CID_EL0
+    mov       x11, #COMP_SIZE
+    mul       x10, x10, x11
+
+    // Load PCC, including function we are jumping to within compartment
+    add       x11, x10, #COMP_OFFSET_PCC
+    ldr       c0, [x19, x11]
+
+    // Load DDC
+    add       x11, x10, #COMP_OFFSET_DDC
+    ldr       c1, [x19, x11]
+
+    // Setup SP
+    mov       x12, sp
+    add       x11, x10, #COMP_OFFSET_STK_ADDR
+    ldr       x11, [x19, x11]
+//    add       x13, x10, #COMP_OFFSET_STK_LEN
```

Comment:
In general we prefer to delete commented code (unless there's a really good reason to keep it around).

Reply:
Whoops. Removed in https://github.com/capablevms/cheri-examples/pull/55/commits/3955d6af194c6f59f5bc9f71b06702e66ecaa9dc.